compiler/rustc_mir_transform/src/ssa.rs

```diff
@@ -228,34 +249,34 @@ impl<'tcx> Visitor<'tcx> for SsaVisitor<'_> {
     }
 
     fn visit_place(&mut self, place: &Place<'tcx>, ctxt: PlaceContext, loc: Location) {
-        if place.projection.first() == Some(&PlaceElem::Deref) {
-            // Do not do anything for storage statements and debuginfo.
+        let location = match ctxt {
+            PlaceContext::MutatingUse(
+                MutatingUseContext::Store | MutatingUseContext::Call | MutatingUseContext::Yield,
+            ) => Some(DefLocation::Body(loc)),
```

Comment:
@tmiasko Following the discussion on https://github.com/rust-lang/rust/pull/116454#discussion_r1348002248, I reuse `DefLocation::Body` for the terminators. Do you agree with this conclusion?

Reply:
So let's go small steps. This PR as-is, then https://github.com/rust-lang/rust/commit/abd4ad1b469d827768d556c305c3d7412ef018f9 in a second PR.